### DISCUSSION TITLE
Support generic type serialization

### DISCUSSION THREAD

**Discussion by mklimko-perpetio** [serverpod/serverpod#509] [Q&A] [answered]
### **Is there way to create generic type serialisation and how to add this to generator?**

For example I have this class:

```
class Either<T1, T2> {
  EitherOption<T1>? left;
  EitherOption<T2>? right;

  Either(this.left, this.right);

  Either.left(dynamic leftData) {
    left = EitherOption(value: leftData);
  }
  Either.right(dynamic rightData) {
    left = EitherOption(value: rightData);
  }

  Future<dynamic> either({
    Function(T1)? leftFunc,
    Function(T2)? rightfunc,
  }) async {
    if (left != null) {
      return await leftFunc?.call(left!.value);
    }
    if (right != null) {
      return await rightfunc?.call(right!.value);
    }
  }

  Either.fromJson(
    Map<String, dynamic> json,
    SerializationManager serializationManager,
  )   : left = EitherOption.fromJson(json["left"]),
        right = EitherOption.fromJson(json["right"]);

  Map<String, dynamic> toJson() => {
        'left': left?.toJson(),
        'right': right?.toJson(),
      };
}
```
either_option.dart

```
class EitherOption<T> {
  final T value;

  EitherOption({required this.value});

  EitherOption.fromJson(Map<String, dynamic> json) : value = json["value"];
  Map<String, dynamic> toJson() => {
        'value': value,
      };
}

```

added to generator like this

```
extraClasses:
  - package:before_lecture_shared/error_handling/either/either.dart:Either
```
and in this case I get **_No deserialization found for type Either<void, CustomException>_**

If I add like this
```
extraClasses:
  - package:before_lecture_shared/error_handling/either/either.dart:Either<T1,T2>
```
then it not generate this class at all.

**Q1: how I can do this if it possible?
Q2: maybe I'll add this feature by contributing in future but how I can do this in alternative way if it is not possible for now?**

**Comment by masreplay** [comment]
@mklimko-perpetio any update?

**Comment by vlidholt** [comment, marked as answer]
I think this is probably not currently supported, but we will look into it for the future!

**Comment by masreplay** [comment]
@vlidholt 
What if I want to make a ```Pagination<T>``` response,
- Do you recommend creating a separate class for each type for example ProductPagination
- Return a map of  ```Map<String, dynamic>``` instead
or there is another way?
